packages/client/src/client/Web/serviceWorkerRegistration.ts

```diff
@@ -53,51 +56,78 @@ export function register(config?: Config) {
         })
       } else {
         // Is not localhost. Just register service worker
-        registerValidSW(swUrl, config)
+        registerServiceWorker(swUrl, config)
       }
     })
   }
 }
 
-function registerValidSW(swUrl: string, config?: Config) {
+function addStateChangeHandler(
```

Comment:
pulled this out as a common handler ... we can use in the usual case when installing worker -> installed, but also when we come across a waiting SW already in "installed" -- this has happened several times in testing (seems to be exacerbated by using slow connection - Slow 3G in Network throttling)